packages/env-utilities/src/lib/env-based-loader.ts

```diff
@@ -0,0 +1,88 @@
+import { config as dotenvConfig, type DotenvConfigOptions, type DotenvConfigOutput, type DotenvParseOutput } from 'dotenv';
+import { expand } from 'dotenv-expand';
+import { existsSync } from 'node:fs';
+import { basename, resolve } from 'node:path';
+
+export interface EnvBasedLoaderOptions extends DotenvConfigOptions {
+	/**
+	 * You may specify a custom environment if `NODE_ENV` isn't sufficient.
+	 */
+	env?: string;
+	/**
+	 * You may specify a required prefix for your dotenv variables (ex. `APP_`).
+	 */
+	prefix?: string;
+}
+
+export function config(options?: EnvBasedLoaderOptions): DotenvConfigOutput {
+	const log = (message: string): void => {
+		options?.debug && console.debug(`[@skyra/env-utilities][DEBUG] ${message}`);
+	};
```

Comment:
Slightly better perf:
```suggestion
	const log = options?.debug
		? (message: string) => console.debug(`[@skyra/env-utilities][DEBUG] ${message}`);
		: (message: string) => undefined;
```